rusk-recovery/src/bin/args.rs

```diff
@@ -0,0 +1,55 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::version;
+use clap::{arg, crate_description, App, AppSettings, ArgMatches};
+use rustc_tools_util::VersionInfo;
+
+pub fn matches() -> ArgMatches {
+    let info = rustc_tools_util::get_version_info!();
+
+    App::new(&info.crate_name)
```

Comment:
Should start using the `clap v3` style of doing things? It's more declarative and, in my opinion, more readable.

Reply:
I'm replying here to your question in the final comment: I didn't use the derive API (I assume you're referring to that) of `clap v3` mostly for two reason:
1. I'm familiar with the builder API, so it take less to write this code instead of learning the new approach.
2. Because exactly like in `v2`, I didn't find a quick way to provide the custom version string, and I always prefer to having the build plus the date in my CLI in addition to the `major.minor.patch`.

So if you know a way to resolve the point 2, we can open a follow up issue to change the way the arguments are written: I put in a separate module so it's easy to modify them without affecting much the rest of the code.

Reply:
We can use a `lazy_static`:

```rust
lazy_static! {
    static ref VERSION: String = {
        let version = env!("CARGO_PKG_VERSION");
        let hash = "somehashhere";

        let mut version = String::from(version);
        version.push('-');
        version.push_str(hash);
        version
    };
}

#[derive(Parser, Debug)]
#[clap(version = &VERSION[..])]
struct Args {
    /// Name of the profile to load.
    #[clap(short, long)]
    profile: String,

    /// SDK server to connect to.
    #[clap(short, long)]
    server: String,
}
```

Reply:
Opened an [issue](https://github.com/dusk-network/rusk/issues/436) about this.